src/main/java/de/tum/in/www1/artemis/service/metrics/MetricsService.java

```diff
@@ -0,0 +1,156 @@
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.repository.metrics.CompetencyMetricsRepository;
+import de.tum.in.www1.artemis.repository.metrics.ExerciseMetricsRepository;
+import de.tum.in.www1.artemis.repository.metrics.LectureUnitMetricsRepository;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.CompetencyInformationDTO;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.CompetencyProgressDTO;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.CompetencyStudentMetricsDTO;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.ExerciseInformationDTO;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.ExerciseStudentMetricsDTO;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.LectureUnitInformationDTO;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.LectureUnitStudentMetricsDTO;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.MapEntryLongLong;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.ResourceTimestampDTO;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.ScoreDTO;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.StudentMetricsDTO;
+
+/**
+ * Service class to access metrics regarding students' learning progress.
+ */
+@Profile(PROFILE_CORE)
+@Service
+public class MetricsService {
+
+    private final ExerciseMetricsRepository exerciseMetricsRepository;
+
+    private final LectureUnitMetricsRepository lectureUnitMetricsRepository;
+
+    private final CompetencyMetricsRepository competencyMetricsRepository;
+
+    public MetricsService(ExerciseMetricsRepository exerciseMetricsRepository, LectureUnitMetricsRepository lectureUnitMetricsRepository,
+            CompetencyMetricsRepository competencyMetricsRepository) {
+        this.exerciseMetricsRepository = exerciseMetricsRepository;
+        this.lectureUnitMetricsRepository = lectureUnitMetricsRepository;
+        this.competencyMetricsRepository = competencyMetricsRepository;
+    }
+
+    /**
+     * Get the metrics for a student in a course.
+     *
+     * @param userId   the id of the student
+     * @param courseId the id of the course
+     * @return the metrics for the student in the course
+     */
+    public StudentMetricsDTO getStudentCourseMetrics(long userId, long courseId) {
+        final var exerciseMetricsDTO = getStudentExerciseMetrics(userId, courseId);
+        final var lectureUnitMetricsDTO = getStudentLectureUnitMetrics(userId, courseId);
+        final var competencyMetricsDTO = getStudentCompetencyMetrics(userId, courseId);
+        return new StudentMetricsDTO(exerciseMetricsDTO, lectureUnitMetricsDTO, competencyMetricsDTO);
+    }
+
+    /**
+     * Get the exercise metrics for a student in a course.
+     *
+     * @param userId   the id of the student
+     * @param courseId the id of the course
+     * @return the metrics for the student in the course
+     */
+    public ExerciseStudentMetricsDTO getStudentExerciseMetrics(long userId, long courseId) {
```

Comment:
could this be done with more efficient database calls? Because we perform 5+ calls here.